### DISCUSSION TITLE
Customizing deserialization of uint64_t

### DISCUSSION THREAD

**Discussion by chfast** [nlohmann/json#3495] [Q&A]
Is there a way to customize how built-in types like `uint64_t` are deserialized by the library? I have a JSON schema where some numeric values are serialized as a string with hex-encoded value, e.g. `"0x8000000000000001"`. I want to make `j.get<uint64_t>()` to work for these too.

**Comment by nlohmann** [comment]
No, this is currently not possible.

**Comment by chfast** [comment]
Can this be done by defining my own serializer as described in https://json.nlohmann.me/features/arbitrary_types/#can-i-write-my-own-serializer-advanced-use ?
